esphome/components/http_request/__init__.py

```diff
@@ -47,6 +47,7 @@
 CONF_MAX_RESPONSE_BUFFER_SIZE = "max_response_buffer_size"
 CONF_ON_RESPONSE = "on_response"
 CONF_HEADERS = "headers"
+CONF_COLLECT_HEADER_NAMES = "collect_header_names"
```

Comment:
Maybe rename to just `collect_headers`, for consistency with the already available headers option?
```suggestion
CONF_COLLECT_HEADERS = "collect_headers"
```

Reply:
That sounds like a good idea - I have done so.